### PR TITLE
feat: add ToolOutputTrimmer for smart context management

### LINKED ISSUE #2467: Built-in ToolOutputTrimmer for call_model_input_filter
## Problem

Agentic applications accumulate large tool outputs (search results, code execution output, error analyses) that consume significant tokens but lose relevance as the conversation progresses. The `call_model_input_filter` hook exists in `RunConfig` but has no built-in implementation — every developer rolls their own trimming logic.

This is related to the broader context management pain described in #177 and #1494.

## Proposal

Add a built-in `ToolOutputTrimmer` class in `agents.memory` that implements `CallModelInputFilter` using a sliding-window approach:

- Walks backward through conversation items counting user messages to find the boundary between "old" and "recent" turns
- Only trims `function_call_output` items in old turns — user messages, assistant messages, and `function_call` items (tool name + arguments) are always preserved
- Replaces large outputs with a compact preview including the tool name and original size
- Skips trimming when the summary would be longer than the original
- Validates configuration parameters at construction time
- Never mutates the original input list

### Usage

```python
from agents import RunConfig
from agents.memory import ToolOutputTrimmer

config = RunConfig(
    call_model_input_filter=ToolOutputTrimmer(
        recent_turns=2,           # keep last 2 user turns intact
        max_output_chars=500,     # trim outputs above 500 chars
        preview_chars=200,        # keep 200-char preview when trimming
        trimmable_tools={"search", "execute_code"},  # optional allowlist
    ),
)
```

## Implementation

I have a working implementation with 30 tests ready as a PR. Key design choices:

- **Dataclass** with `__call__` implementing the `CallModelInputFilter` protocol
- **`frozenset`** for `trimmable_tools` (immutable; coerces sets/lists automatically)
- **`TYPE_CHECKING`** import to avoid circular dependency with `run_config.py`
- **`debug`-level logging** to avoid noise in production

### PR DESCRIPTION
**Pull request by jkrish**
## Summary

Adds a built-in `ToolOutputTrimmer` class that implements `CallModelInputFilter` — a sliding-window filter that surgically trims large tool outputs from older conversation turns while keeping recent turns at full fidelity.

Closes #2467. Related: #177, #1494.

### What it does

- Derives turn boundaries from user messages (no explicit turn structure needed)
- Only targets `function_call_output` items in old turns — preserves user/assistant messages and `function_call` items
- Resolves tool names via `call_id` mapping for informative summaries
- Skips trimming when the summary would exceed the original size
- Validates all parameters at construction (`recent_turns >= 1`, `max_output_chars >= 1`, `preview_chars >= 0`)
- Never mutates the original input list

### Usage

```python
from agents import RunConfig
from agents.memory import ToolOutputTrimmer

config = RunConfig(
    call_model_input_filter=ToolOutputTrimmer(
        recent_turns=2,
        max_output_chars=500,
        preview_chars=200,
        trimmable_tools={"search", "execute_code"},
    ),
)
```

### Files changed

| File | Change |
|------|--------|
| `src/agents/memory/tool_output_trimmer.py` | New — `ToolOutputTrimmer` dataclass (170 lines) |
| `src/agents/memory/__init__.py` | Export `ToolOutputTrimmer` |
| `src/agents/__init__.py` | Export `ToolOutputTrimmer` |
| `tests/memory/test_tool_output_trimmer.py` | 30 tests across 6 test classes |

## Test plan

- [x] `make format` — clean
- [x] `make lint` — clean
- [x] `make mypy` — 0 new errors
- [x] `make tests` — 1888 passed, 4 skipped, 0 failures (no regressions)
- [x] 30 dedicated tests covering defaults, validation, boundary detection, trimming behavior, sliding window, and edge cases

### CODE REVIEW

**Comment by seratch** [comment]
Thanks for sharing this idea. I haven't checked in depth yet, but we'll consider this in the next minor version.

File: src/agents/extensions/tool_output_trimmer.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
+"""Built-in call_model_input_filter that trims large tool outputs from older turns.
```
Comment (seratch) [inline src/agents/extensions/tool_output_trimmer.py:1]: can you move this to extensions?

Comment (jkrish) [reply src/agents/extensions/tool_output_trimmer.py:1]: Done — moved to `src/agents/extensions/memory/tool_output_trimmer.py` with updated imports and tests. Import path is now `from agents.extensions.memory import ToolOutputTrimmer`.

Comment (seratch) [reply src/agents/extensions/tool_output_trimmer.py:1]: I don't think call_model_input_filter is only for memory feature users. So, you can place it just under extensions module.

Comment (jkrish) [reply src/agents/extensions/tool_output_trimmer.py:1]: Done — moved up to `src/agents/extensions/tool_output_trimmer.py`. Import path is now:

```python
from agents.extensions import ToolOutputTrimmer
```

**Review by seratch — APPROVED** [review]
